### PR TITLE
Mark valid scenes with component

### LINKED ISSUE #9115: Emit event when scene is ready
## What problem does this solve or what need does it fill?

Scenes spawned with delay and in order to check if scene is spawned we use [SceneSpawner::instance_is_ready](https://docs.rs/bevy/latest/bevy/scene/struct.SceneSpawner.html#method.instance_is_ready). But it's not very convenient because you can't actually react on spawning.

## What solution would you like?

Introduce `ReadyScene` component that inserted after scene spawning.

## What alternative(s) have you considered?

Events, but marker component is more flexible because we also be able to filter loaded scenes in queries.

### PR DESCRIPTION
**Pull request by Olle-Lukowski**
# Objective

Fixes #9115 

## Solution

Added a ValidScene Component, as the issue suggests.
---

### CODE REVIEW

**Comment by Shatur** [comment]
You need to insert it to scene entity. And remove when scene becomes despawned / re-loaded.

**Comment by mockersf** [comment]
marking as controversial until the discussion in the issue is resolved

File: crates/bevy_scene/src/scene_spawner.rs [hunk 1]
```diff
@@ -28,6 +29,9 @@ impl InstanceId {
     }
 }
 
+#[derive(Component)]
+pub struct ValidScene;
```
Comment (alice-i-cecile) [inline crates/bevy_scene/src/scene_spawner.rs:33]: This needs doc strings.

**Comment by alice-i-cecile** [comment]
Closed in favor #9115.

**Pull request closed without merging**